src/main/scala/com/databricks/labs/overwatch/pipeline/Restore.scala

```diff
@@ -0,0 +1,129 @@
+ * @param _database           Workspace Database Name.
+ * @param _config             Source Workspace Config.
+ */
+
+class Restore (_sourceETLDB: String, _targetPrefix: String, _workspace: Workspace, _database: Database, _config: Config)
+  extends Pipeline(_workspace, _database, _config) {
+
+  import spark.implicits._
+
+  private val targetPrefix = removeTrailingSlashes(_targetPrefix)
+  private val workSpace = _workspace
+
+  private val logger: Logger = Logger.getLogger(this.getClass)
+
+  private[overwatch] def restore(
+                                  allTargets: Array[PipelineTable],
+                                  cloneLevel: String = "DEEP",
+                                ): Unit = {
+    val acceptableCloneLevels = Array("DEEP", "SHALLOW")
+    require(acceptableCloneLevels.contains(cloneLevel.toUpperCase), s"SNAP CLONE ERROR: cloneLevel provided is " +
+      s"$cloneLevel. CloneLevels supported are ${acceptableCloneLevels.mkString(",")}.")
+
+    try {
+      dbutils.fs.ls(targetPrefix)
+    } catch {
+      case e: Exception => dbutils.fs.mkdirs(targetPrefix)
+    }
+
+    try {
+      if (dbutils.fs.ls(targetPrefix).isEmpty) {
+        val cloneSpecs = new Snapshot(workSpace.getConfig.databaseName, targetPrefix, workspace, workspace.database,
+          workspace.getConfig, "restore").buildCloneSpecs(cloneLevel, allTargets)
+        val cloneReport = Helpers.parClone(cloneSpecs)
+        val restoreReportPath = s"${targetPrefix}/restore_report/"
+        logger.log(Level.INFO,s"Restoration is successful. Please check restoreReportPath ${restoreReportPath} for more details")
+        cloneReport.toDS.write.format("delta").mode("append").save(restoreReportPath)
+      } else {
+        logger.log(Level.INFO, "Target Path is not Empty...... Could not proceed with Restoration")
+      }
+
+    } catch {
+      case e: Throwable =>
+        val failMsg = PipelineFunctions.appendStackStrace(e)
+        logger.log(Level.WARN, failMsg)
+    }
+  }
+}
+
+object Restore extends SparkSessionWrapper {
+  private val logger: Logger = Logger.getLogger(this.getClass)
+
+    def apply(
+               workSpace: Workspace,
+               allTarget: Array[PipelineTable],
+               targetPrefix : String,
+               backupPath: String,
+               cloneLevel: String
+             ): Unit = {
+
+      val restoration = new Restore(workSpace.getConfig.databaseName, targetPrefix, workSpace, workSpace.database, workSpace.getConfig)
+      restoration.restore(allTarget,cloneLevel)
+
+    }
+
+    /**
+     * Create a backup of the Overwatch datasets
+     *
+     * @param arg(0)        Source ETL Path Prefix from where restore need to be performed
+     * @param arg(1)        Target ETL Path Prefix to where restore data would be loaded.
+     * @return
+     */
+
+    def main(args: Array[String]): Unit = {
+
+      val sourcePrefix = args(0)
+      val orgID = Initializer.getOrgId
+      val targetPrefix = args(1)
```

Comment:
1)If we we don't provide args(0) or args(1) how we are performing exception handling?
2)If we provide wrong args then how we will handle the exceptions?

Reply:
Done.